Source/CesiumRuntime/Private/Tests/CesiumLoadTest.cpp

```diff
@@ -299,15 +304,93 @@ bool RunLoadTest(const size_t locationIndex) {
         loadElapsedTime);
   }
 
+  if (afterTest) {
+    afterTest(context);
+  }
+
   return !timedOut;
 }
 
 bool FCesiumLoadTestDenver::RunTest(const FString& Parameters) {
-  return RunLoadTest(1);
+  return RunLoadTest(setupForDenver);
 }
 
 bool FCesiumLoadTestGoogleplex::RunTest(const FString& Parameters) {
-  return RunLoadTest(0);
+  return RunLoadTest(setupForGoogleTiles);
+}
+
+bool FCesiumLoadTestMontrealPointCloud::RunTest(const FString& Parameters) {
+  auto setup = [this](LoadTestContext& context) {
+    FVector targetOrigin(-73.616526, 45.57335, 95.048859);
+    FString ionToken(
+        "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJqdGkiOiI2NmZhZTk4NS01MDFmLTRjODgtOTlkYy04NjIwODhiZWExOGYiLCJpZCI6MjU5LCJpYXQiOjE2ODg1MTI4ODd9.haoe5hsJyfHk1dQAHVK6N8dW_kfmtdbyuhlGwFdEHbM");
+
+    FCesiumCamera camera;
+    camera.ViewportSize = FVector2D(1024, 768);
+    camera.Location = FVector(0, 0, 0);
+    camera.Rotation = FRotator(-90.0, 0.0, 0.0);
+    camera.FieldOfViewDegrees = 90;
+    context.setCamera(camera);
+
+    context.georeference->SetGeoreferenceOriginLongitudeLatitudeHeight(
+        targetOrigin);
+
+    context.pawn->SetActorLocation(FVector(0, 0, 0));
+    context.pawn->SetActorRotation(FRotator(-90.0, 0.0, 0.0));
+
+    // Montreal Point Cloud
+    ACesium3DTileset* montrealTileset =
+        context.world->SpawnActor<ACesium3DTileset>();
+    montrealTileset->SetTilesetSource(ETilesetSource::FromCesiumIon);
+    montrealTileset->SetIonAssetID(28945);
+    montrealTileset->SetIonAccessToken(ionToken);
+    montrealTileset->SetMaximumScreenSpaceError(16.0);
+    montrealTileset->SetActorLabel(TEXT("Montreal Point Cloud"));
+
+    context.tilesets.push_back(montrealTileset);
+  };
+
+  auto after = [this](LoadTestContext& context) {
+    // Zoom way out
+    FCesiumCamera zoomedOut;
+    zoomedOut.ViewportSize = FVector2D(1024, 768);
+    zoomedOut.Location = FVector(0, 0, 7240000.0);
+    zoomedOut.Rotation = FRotator(-90.0, 0.0, 0.0);
+    zoomedOut.FieldOfViewDegrees = 90;
+    context.setCamera(zoomedOut);
+
+    context.pawn->SetActorLocation(zoomedOut.Location);
+
+    context.setSuspendUpdate(false);
+    tickWorldUntil(context, 10, breakWhenTilesetsLoaded);
+    context.setSuspendUpdate(true);
+
+    Cesium3DTilesSelection::Tileset* pTileset =
+        context.tilesets[0]->GetTileset();
+    if (TestNotNull("Tileset", pTileset)) {
```

Comment:
So were you going the direction of combining a performance test with tests for correctness?
I viewed these as somewhat separate, thus the Cesium.Performance.X namespace.
But even if we make them separate, we should reuse as much code as possible.
(I like this camera reposition test btw)

Reply:
Well.... yeah, but it wasn't a very carefully considered choice. I was just trying to chase down a bug, and the performance framework gave me an easy way to do that, even if I was abusing it.

I think assertions of correctness _do_ have a place in performance tests, though. Mostly as sanity checks: is that performance test fast because we optimized something awesomely? Or is it fast because we broke something and produces incorrect results (quickly)?